src/engine.rs

```diff
@@ -356,41 +353,51 @@ impl Reedline {
                 // pasting text, resizes, blocking this thread (e.g. during debugging)
                 // We should be able to handle all of them as quickly as possible without causing unnecessary output steps.
                 while event::poll(Duration::from_millis(POLL_WAIT))? {
-                    // TODO: Maybe replace with a separate function processing the buffered event
                     match event::read()? {
                         Event::Resize(x, y) => {
                             latest_resize = Some((x, y));
                         }
-                        x => crossterm_events.push(x),
+                        enter @ Event::Key(KeyEvent {
```

Comment:
Yeah it will leave all events after the enter, hit the validator (sadly repaint in this config) and either return `Reedline::read_line` to the caller or continue to drain until the next enter if the line is not a valid expression

Reply:
Very nice. It handles paste way better this way. Lets merge it to play with it

Reply:
I think I have a solution for the unnecessary repaints. Let me run some checks

Reply:
Solution is suboptimal for short lines but should accelerate pasting the dump from https://github.com/nushell/reedline/issues/183